Terminal.Gui/Views/Menu/MenuBar.cs

```diff
@@ -133,16 +133,24 @@ public MenuBar ()
                                                   {
                                                       CloseOtherOpenedMenuBar ();
 
-                                                      return Select (Menus.IndexOf (ctx.KeyBinding?.Context));
+                                                      if (ctx is not CommandContext<KeyBinding> keyCommandContext)
+                                                      {
+                                                          return false;
+                                                      }
+                                                      return Select (Menus.IndexOf (keyCommandContext.Binding.Data));
                                                   });
         AddCommand (Command.Select, ctx =>
                                     {
-                                        if (ctx.Data is MouseEventArgs)
+                                        if (ctx is not CommandContext<KeyBinding> keyCommandContext)
+                                        {
+                                            return false ;
+                                        }
+                                        if (keyCommandContext.Binding.Data is MouseEventArgs)
                                         {
                                             // HACK: Work around the fact that View.MouseClick always invokes Select
                                             return false;
                                         }
-                                        var res = Run ((ctx.KeyBinding?.Context as MenuItem)?.Action!);
+                                        var res = Run ((keyCommandContext.Binding.Data as MenuItem)?.Action!);
```

Comment:
Is Data the owning view? Or arbitrary object? Could be named better?

Reply:
```cs
    /// <summary>
    ///     Arbitrary context that can be associated with this key binding.
    /// </summary>
    public object? Data { get; set; }
```

The ComboBox case is the easiest to understand: It has an arbitrary # of sub-items that are not sub-Views.  Each has a HotKey that is enabled for the ComboBox, in addition to the primary HotKey (from `Title`). When the key bindings are set, `Data` is set to the index of the radioitem, so that when an item HotKey is pressed, the RadioGroup gets an Invoke and can do this:

![image](https://github.com/user-attachments/assets/cbfbaad0-6c76-4873-aabb-792f5a5a0169)

Reply:
Got it, it works.  Just involves a bit of hidden magic whenever I see `object` but I think this is ok now you have explained thankyou.